common/common.go

```diff
@@ -0,0 +1,366 @@
+}
+
+type influxHandleType struct {
+	cl client.Client
+}
+
+func (q *influxHandleType) Query(queryStr, database, epoch string) (
+	*client.Response, error) {
+	aQuery := client.NewQuery(queryStr, database, epoch)
+	return q.cl.Query(aQuery)
+}
+
+func (q *influxHandleType) Close() error {
+	return q.cl.Close()
+}
+
+type handleCreaterType func(addr string) (handleType, error)
+
+func influxCreateHandle(addr string) (handleType, error) {
+	cl, err := client.NewHTTPClient(client.HTTPConfig{Addr: addr})
+	if err != nil {
+		return nil, err
+	}
+	return &influxHandleType{cl: cl}, nil
+}
+
+func getConcurrentResponses(
+	endpoints []queryerType,
+	queries []*influxql.Query,
+	epoch string,
+	logger *log.Logger) (
+	*client.Response, error) {
+	if len(endpoints) != len(queries) {
+		panic("endpoints and queries parameters must have same length")
+	}
+	// These are placeholders for the response and error from each influx db
+	// instance.
+	responseList := make([]*client.Response, len(queries))
+	errs := make([]error, len(queries))
+
+	var wg sync.WaitGroup
+	for i, query := range queries {
+		// Query not applicable, skip
+		if query == nil {
+			continue
+		}
+		wg.Add(1)
+		go func(
+			n queryerType,
+			query string,
+			responseHere **client.Response,
+			errHere *error) {
+			*responseHere, *errHere = n.Query(query, epoch)
+			wg.Done()
+		}(endpoints[i],
+			query.String(),
+			&responseList[i],
+			&errs[i])
+	}
+	wg.Wait()
+
+	// These will be the responses from influx servers that we merge
+	var responsesToMerge []*client.Response
+
+	// In case none of the responses are viable, report this error
+	// back to client
+	var lastErrorEncountered error
+
+	for i := range queries {
+		if queries[i] == nil {
+			continue
+		}
+		err := errs[i]
+		response := responseList[i]
+		if err == nil && response.Error() != nil {
+			err = response.Error()
+		}
+		if err == nil {
+			responsesToMerge = append(responsesToMerge, response)
+		} else {
+			if logger != nil {
+				logger.Println(err)
+			}
+			lastErrorEncountered = err
+		}
+	}
+	// errors but no viable responses
+	if len(responsesToMerge) == 0 && lastErrorEncountered != nil {
+		return nil, lastErrorEncountered
+	}
+	return responses.Merge(responsesToMerge...)
+}
+
+func newInfluxForTesting(
+	influx config.Influx, creater handleCreaterType) (*Influx, error) {
+	handle, err := creater(influx.HostAndPort)
+	if err != nil {
+		return nil, err
+	}
+	return &Influx{data: influx, handle: handle}, nil
+}
+
+func newInfluxListForTesting(
+	influxes config.InfluxList, creater handleCreaterType) (
+	*InfluxList, error) {
+	if len(influxes) == 0 {
+		return nil, nil
+	}
+	influxes = influxes.Order()
+	result := &InfluxList{instances: make([]*Influx, len(influxes))}
+	for i := range influxes {
+		var err error
+		result.instances[i], err = newInfluxForTesting(influxes[i], creater)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return result, nil
+}
+
+func (l *InfluxList) _close() error {
```

Comment:
Although close is not a reserved word in go, go has the close built-in function. When I try to use lowercase close() as a method name, my editor highlights it as a built in function. To avoid the confusion, I used "_close"